x-pack/elastic-agent/spec/fleet-server.yml

```diff
@@ -0,0 +1,61 @@
+name: Fleet Server
+cmd: fleet-server
+args: ["--agent-mode"]
+artifact: fleet-server
+rules:
+  - fix_stream: {}
+
+  - filter_values:
+      selector: inputs
+      key: type
+      values:
+        - fleet-server
+
+  - filter_values:
+      selector: inputs
+      key: enabled
+      values:
+        - true
+
+  - remove_key:
+      key: output
+
+  - select_into:
```

Comment:
Having the output directly as part of the input would also simplify the spec file here?

Reply:
It would still need to be moved because the output is a top-level key to Fleet Server.

I also don't think we want that, because we do not want to send username/password from the input. That also would have a chicken-n-egg problem of getting that information initially.

Reply:
This goes back to a previous discussion. I think the fleet-server config should be designed and optimised for what is needed for Elastic Agent. So either it directly supports input as the default or it maps it internally. But the translation should not be up to the Elastic Agent. It can be a temporary solution for now.

Not sure I follow the chicken-egg part around username / password? Can you elaborate?

Reply:
Fleet Server does all of that. It has a top level `inputs` just like Elastic Agent. The issue is the output, which is also top level `output` and matches Elastic Agent.

The issue is on start-up there is no output, because Elastic Agent has not enrolled but Fleet Server needs to connect to an output so that the Elastic Agent can enroll.